modin/pandas/test/test_io.py

```diff
@@ -1622,7 +1622,9 @@ def test_HDFStore(self):
 
             modin_store.close()
             pandas_store.close()
-            assert assert_files_eq(unique_filename_modin, unique_filename_pandas)
```

Comment:
Here, most likely the same problem with duplicate data in case of worker crash: https://github.com/modin-project/modin/issues/3659. In this way, we will simply hide the error.

Reply:
Do you mean that the tests are flaky currently because the ray workers sometimes crash while writing these files to disk? I would expect that to cause the checks I'm adding on the `foo` dataframe to fail because the test wouldn't be able to parse the file.

Github says you are requesting a change, but I don't see a change. What do you propose instead to fix the flakiness?

Reply:
@anmyachev There are no worker crashes in this particular test failure, it is a quirk of HDF5 file format. I think this change is a good one, we just need to make sure that the tests will pass now.

Reply:
ok

Reply:
Why do we read data using pandas but not Modin in this line?

Reply:
I used pandas to read the CSV here because this part of the test was checking that modin writes the HDF file correctly, not that it reads the HDF file correctly.

Reply:
I am just wondering why pandas is used to read the data here but below (on 1636 line) Modin is used to read the data?

Reply:
Here, the test checks that Modin correctly **writes** an HDF file. It is not checking that Modin reads HDF files correctly. We need to check that the HDF file has been written correctly whether we use Modin or pandas, so we use pandas to read both files.

In line 1636, the test checks that Modin correctly **reads** an HDF file. It is not checking that Modin writes HDF files correctly. We first write a file with pandas, then read it with both Modin and pandas and check that we get the same result.

Reply:
I see, makes sense. @mvashishtha , thanks!